packages/Discretization/src/DTK_PointSearch_def.hpp

```diff
@@ -307,39 +307,62 @@ std::tuple<Kokkos::View<int *, DeviceType>, Kokkos::View<int *, DeviceType>,
 PointSearch<DeviceType>::getSearchResults()
 {
     // Flatten the results
+    using ExecutionSpace = typename DeviceType::execution_space;
     unsigned int n_ref_pts = 0;
     for ( unsigned int topo_id = 0; topo_id < DTK_N_TOPO; ++topo_id )
         n_ref_pts += _reference_points[topo_id].extent( 0 );
 
     Kokkos::View<int *, DeviceType> ranks( "ranks", n_ref_pts );
-    Kokkos::View<int *, DeviceType> cell_indices( "cell_indices", n_ref_pts );
-    Kokkos::View<Point *, DeviceType> ref_pts( "ref_pts", n_ref_pts );
-    Kokkos::View<unsigned int *, DeviceType> query_ids( "query_ids",
-                                                        n_ref_pts );
-
     Kokkos::deep_copy( ranks, _comm->getRank() );
-    auto cell_indices_host = Kokkos::create_mirror_view( cell_indices );
-    Kokkos::deep_copy( cell_indices_host, cell_indices );
-    auto ref_pts_host = Kokkos::create_mirror_view( ref_pts );
-    Kokkos::deep_copy( ref_pts_host, ref_pts );
-    auto query_ids_host = Kokkos::create_mirror_view( query_ids );
-    Kokkos::deep_copy( query_ids_host, query_ids );
-    unsigned int pos = 0;
+    Kokkos::View<int *, DeviceType> flatten_cell_indices(
+        "flatten_cell_indices", n_ref_pts );
+    auto flatten_cell_indices_host =
+        Kokkos::create_mirror_view( flatten_cell_indices );
+    Kokkos::View<unsigned int *, DeviceType> flatten_query_ids(
+        "flatten_query_ids", n_ref_pts );
+    Kokkos::View<Point *, DeviceType> flatten_ref_pts( "flatten_ref_pts",
+                                                       n_ref_pts );
+    unsigned int n_copied_pts = 0;
     for ( unsigned int topo_id = 0; topo_id < DTK_N_TOPO; ++topo_id )
     {
         unsigned int const size = _query_ids[topo_id].extent( 0 );
+
+        // First fill cell_indices. This has to be done on the host because
+        // _cell_indices_map only exists on the host.
```

Comment:
Good catch. How do you realize it?

Reply:
I was thinking that I had a reason not to use `parallel_for` when I wrote this the first time. So I checked for data that only leaved on the host 🙂